scripts/python/check_jbod_psu_status.py

```diff
@@ -0,0 +1,87 @@
+from integralstor import command, alerts
+import re, sys
+
+def get_enclosure_devices():
+    # Returns a list of /dev/sgX path for Connected Storage Enclosures
+    enclosures = []
+    try:
+        sg_devs, err = command.get_command_output('sg_map -i')
+        if err:
+            raise Exception(err)
+
+        if sg_devs:
+            for sg_dev in sg_devs:
+                    encl = re.findall('ENCL', sg_dev, re.IGNORECASE)
+                    if encl:
+                        res = re.findall(r'^\/\w+\/\w+', sg_dev)
+                        enclosures += res
+    except Exception, e:
+        return None, "Error retrieving enclosures : %s" % str(e)
+    else:
+        return enclosures, None
+
+def get_psu_status():
+    psu_status = {}
+    try:
+        encls, err = get_enclosure_devices()
+        if err:
+            raise Exception(err)
+
+        if encls:
+            for encl in encls:
+                for psu in range(2):
+                    tmp, err = command.get_command_output('sg_ses --index=ps,%d %s' % (psu, encl))
```

Comment:
What happens if there is only one PSU? Will it raise an exception or return None?

Reply:
It should raise an exception, I don't have a setup right now to test it out.